iroh-p2p/src/rpc.rs

```diff
@@ -188,36 +165,57 @@ impl RpcP2p for P2p {
     }
 
     #[tracing::instrument(skip(self, req))]
-    async fn fetch_provider_dht(
-        &self,
-        req: ProviderKey,
-    ) -> Result<Pin<Box<dyn Stream<Item = Result<Providers>> + Send>>> {
-        let cid: Cid = req.key.clone().try_into()?;
+    fn fetch_provider_dht(
+        self,
+        req: FetchProvidersDhtRequest,
+    ) -> BoxStream<'static, RpcResult<FetchProvidersDhtResponse>> {
+        async move {
+            let stream = self.fetch_provider_dht0(req).await?;
+            Ok(stream.map(|x| x.map_err(RpcError::from)))
+        }
+        .try_flatten_stream()
+        .boxed()
+    }
+
+    #[tracing::instrument(skip(self, req))]
+    async fn fetch_provider_dht0(
```

Comment:
why are there two methods?

Reply:
One is boilerplate to adapt the function signature to what server_streaming expects. Added a comment.